packages/SwingSet/src/initializeSwingset.js

```diff
@@ -157,45 +159,57 @@ function resolveSpecFromConfig(dirname, specPath) {
  * path and make sure it has a `parameters` property if it's supposed to.
  *
  * @param {SwingSetConfigDescriptor | void} desc  The config descriptor to be normalized.
- * @param {string} dirname  The pathname of the directory in which the config file was found
+ * @param {string} referrer  The pathname of the file or directory in which the
+ * config file was found
  * @param {boolean} expectParameters `true` if the entries should have parameters (for
  *    example, `true` for `vats` but `false` for bundles).
  */
-function normalizeConfigDescriptor(desc, dirname, expectParameters) {
+async function normalizeConfigDescriptor(desc, referrer, expectParameters) {
+  const normalizeSpec = async (entry, key) => {
+    return resolveSpecFromConfig(referrer, entry[key]).then(spec => {
+      entry[key] = spec;
+    });
+  };
+
+  const jobs = [];
   if (desc) {
     for (const name of Object.keys(desc)) {
       const entry = desc[name];
       if ('sourceSpec' in entry) {
-        entry.sourceSpec = resolveSpecFromConfig(dirname, entry.sourceSpec);
+        jobs.push(normalizeSpec(entry, 'sourceSpec'));
       }
       if ('bundleSpec' in entry) {
-        entry.bundleSpec = resolveSpecFromConfig(dirname, entry.bundleSpec);
+        jobs.push(normalizeSpec(entry, 'bundleSpec'));
       }
       if (expectParameters && !entry.parameters) {
         entry.parameters = {};
       }
     }
   }
+  return Promise.all(jobs);
 }
 
 /**
  * Read and parse a swingset config file and return it in normalized form.
  *
  * @param {string} configPath  Path to the config file to be processed
  *
- * @returns {SwingSetConfig | null} the contained config object, in normalized form, or null if the
+ * @returns {Promise<SwingSetConfig | null>} the contained config object, in normalized form, or null if the
  *    requested config file did not exist.
  *
  * @throws {Error} if the file existed but was inaccessible, malformed, or otherwise
  *    invalid.
  */
-export function loadSwingsetConfigFile(configPath) {
+export async function loadSwingsetConfigFile(configPath) {
   try {
     const config = JSON.parse(fs.readFileSync(configPath, 'utf-8'));
-    const dirname = path.dirname(configPath);
-    normalizeConfigDescriptor(config.vats, dirname, true);
-    normalizeConfigDescriptor(config.bundles, dirname, false);
-    // normalizeConfigDescriptor(config.devices, dirname, true); // TODO: represent devices
+    const referrer = new URL(
+      configPath,
+      `file:///${process.cwd()}/`,
+    ).toString();
+    await normalizeConfigDescriptor(config.vats, referrer, true);
+    await normalizeConfigDescriptor(config.bundles, referrer, false);
```

Comment:
Oddly, the linter doesn’t spot this opportunity to do these normalization jobs concurrently, just because they’re not in a loop.